### ISSUE TITLE
Cannot install app on iPad

### ISSUE THREAD

**Issue by mbcoder17** [SquaredTiki/Dringend-Issues#48]
With the new update my projects finally compile, but when I try to run them they won't install on my ipad. I get this error:

![image](https://cloud.githubusercontent.com/assets/4973557/2572487/253ea1ac-b914-11e3-80ff-a427b4bef0a9.jpg)

**Comment by SquaredTiki** [comment]
Have you made sure that you've updated The Constructor on your Mac to version 1.0.1?

**Comment by mbcoder17** [comment]
Yes

**Comment by SquaredTiki** [comment]
Are you definitely using a provisioning profile (developer not distribution) which is installed on the device?

**Comment by mbcoder17** [comment]
Yes, I'm using the team profile....

**Comment by SquaredTiki** [comment]
Good. Could you check and copy here your device logs for the period of time whilst the app is compiling and then installation is attempted?

**Comment by mbcoder17** [comment]
Sure, gimme one minute...

**Comment by mbcoder17** [comment]
Here ya go! The device locked at some point, but I don't think that has anything to do with it not installing:

```
Mar 31 16:53:07 Jacobs-iPad Dringend[2095] <Warning>: <Commander> Zip for project will be located at '/private/var/mobile/Applications/2F39DB61-9A7E-40E0-9463-2B9EB772DA1E/tmp/project.zip'
Mar 31 16:53:07 Jacobs-iPad Dringend[2095] <Warning>: <Commander> sending header dictionary: {
        buildOptions =     {
            destination = "generic/platform=iOS";
            scheme = "My App";
        };
        buildType = 1;
        dataLength = 10037;
        exportOptions =     {
            exportFormat = IPA;
        };
    }
Mar 31 16:53:07 Jacobs-iPad Dringend[2095] <Warning>: <Commander> connected to host
Mar 31 16:53:07 Jacobs-iPad Dringend[2095] <Warning>: <Commander> wrote data with tag 33
Mar 31 16:53:07 Jacobs-iPad Dringend[2095] <Warning>: <Commander> wrote data with tag 0
Mar 31 16:53:07 Jacobs-iPad Dringend[2095] <Warning>: <Commander> wrote data with tag 343
Mar 31 16:53:07 Jacobs-iPad Dringend[2095] <Warning>: <Commander> wrote data with tag 0
Mar 31 16:53:08 Jacobs-iPad Dringend[2095] <Warning>: <Commander> Did received return data
Mar 31 16:53:08 Jacobs-iPad Dringend[2095] <Warning>: <Commander> was notified that socket connection was accepted and data was received
Mar 31 16:53:15 Jacobs-iPad lockdownd[25] <Notice>: 001ac000 set_response_error: handle_get_value GetProhibited
Mar 31 16:53:15 Jacobs-iPad lockdownd[25] <Notice>: 0050c000 special_case_get: MGCopyAnswer(kMGQReleaseType) returned NULL
Mar 31 16:53:15 Jacobs-iPad lockdownd[25] <Notice>: 0050c000 special_case_get: MGCopyAnswer(kMGQSDIOProductInfo) returned NULL
Mar 31 16:53:15 Jacobs-iPad lockdownd[25] <Notice>: 0050c000 special_case_get: MGCopyAnswer(kMGQSDIOManufacturerTuple) returned NULL
Mar 31 16:53:15 Jacobs-iPad lockdownd[25] <Notice>: 0050c000 special_case_get: MGCopyAnswer(kMGQBasebandBoardSnum) returned NULL
Mar 31 16:53:15 Jacobs-iPad lockdownd[25] <Notice>: 0050c000 special_case_get: MGCopyAnswer(kMGQWirelessBoardSnum) returned NULL
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2193] <Notice>: Found paths:
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2193] <Notice>: /usr/lib/dyld
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2193] <Notice>: /System/Library/Caches/com.apple.dyld/dyld_shared_cache_arm64
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2193] <Notice>: /System/Library/Caches/com.apple.dyld/dyld_shared_cache_armv7s
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2194] <Notice>: Found paths:
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2194] <Notice>: /usr/lib/dyld
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2194] <Notice>: /System/Library/Caches/com.apple.dyld/dyld_shared_cache_arm64
Mar 31 16:53:17 Jacobs-iPad DTFetchSymbols[2194] <Notice>: /System/Library/Caches/com.apple.dyld/dyld_shared_cache_armv7s
Mar 31 16:53:21 Jacobs-iPad touchsetupd[126] <Warning>: WPTransfer deallocing
Mar 31 16:53:21 Jacobs-iPad wirelessproxd[2108] <Notice>: (Error) error event: (<OS_xpc_error: <error: 0x19176fdc8> { count = 1, contents =
        "XPCErrorDescription" => <string: 0x1917700d0> { length = 18, contents = "Connection invalid" }
    }>)
Mar 31 16:53:51 Jacobs-iPad assistantd[1595] <Notice>: connection 19 total: 0, connected: 0, subflow state changed, removing subflow [cid=1, state=Disconnected, ifindex=4, errno = 53 (Software caused connection abort)]
Mar 31 16:53:51 Jacobs-iPad assistantd[1595] <Error>: tcp_connection_destination_subflow_add MPTCP_ERROR: 19 connectx failed for [cid=2, state=Allocated, ifindex=4]: 53 - Software caused connection abort
Mar 31 16:54:16 Jacobs-iPad locationd[58] <Notice>: need a scan, count, 0, 0, lwatchdog, 0.0, interval, 60.0, needWatchdog, 1
Mar 31 16:54:16 Jacobs-iPad locationd[58] <Notice>: Fence: loc watchdog cancel, count, 1, 0, client, 0x0
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> Did received return data
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> did read return header from socket
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> Did received return data
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> did read export data from socket
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> Received output and export data with name Export.ipa and byte length 42240
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> Zip file written to path: /private/var/mobile/Applications/2F39DB61-9A7E-40E0-9463-2B9EB772DA1E/tmp/com.squaredtiki.the-constructor/38C49720-B4CE-4FA2-9671-426A01616A18/Export.zip, resultant byte length: 42240
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> Export file unzipped to path: /private/var/mobile/Applications/2F39DB61-9A7E-40E0-9463-2B9EB772DA1E/tmp/com.squaredtiki.the-constructor/38C49720-B4CE-4FA2-9671-426A01616A18/Export.ipa, resultant byte length: 46125
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: IPA copied to: /private/var/mobile/Applications/2F39DB61-9A7E-40E0-9463-2B9EB772DA1E/tmp/1FF171BA-148B-4FF6-9BDA-BB961ED448EC/ExportedApp.ipa
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: Plist written temporarily to: /private/var/mobile/Applications/2F39DB61-9A7E-40E0-9463-2B9EB772DA1E/tmp/ExportedApp.plist
Mar 31 16:54:23 Jacobs-iPad Dringend[2095] <Warning>: <Commander> did disconnect with error (null)
Mar 31 16:54:26 Jacobs-iPad Dringend[2095] <Warning>: Plist uploaded to dropbox with (hopefully direct) URL: https://dl.dropboxusercontent.com/s/63xn368uktmxk52/7E68DCA2-3E1A-4D06-B978-67AE44602CCB-ExportedApp.plist
Mar 31 16:54:26 Jacobs-iPad Dringend[2095] <Warning>: HTML written to: /private/var/mobile/Applications/2F39DB61-9A7E-40E0-9463-2B9EB772DA1E/tmp/1FF171BA-148B-4FF6-9BDA-BB961ED448EC/index.html
Mar 31 16:54:26 Jacobs-iPad Dringend[2095] <Warning>: Setting document root: /private/var/mobile/Applications/2F39DB61-9A7E-40E0-9463-2B9EB772DA1E/tmp/1FF171BA-148B-4FF6-9BDA-BB961ED448EC
Mar 31 16:54:26 Jacobs-iPad Dringend[2095] <Warning>: Started HTTP Server on port 55993
Mar 31 16:54:27 Jacobs-iPad itunesstored[2126] <Error>: TASK-ASSERT: cfurlcache - ProcessCacheTask - FAILED to get task-assertion, going commando with 1 items to process.
Mar 31 16:54:40 Jacobs-iPad locationd[58] <Notice>: Location icon should now be in state 'Active'
Mar 31 16:54:40 Jacobs-iPad SpringBoard[17] <Error>: TASK-ASSERT: cfurlcache - ProcessCacheTask - FAILED to get task-assertion, going commando with 1 items to process.
Mar 31 16:54:51 Jacobs-iPad locationd[58] <Notice>: Location icon should now be in state 'Region Monitoring'
Mar 31 16:56:13 Jacobs-iPad backboardd[31] <Notice>: ALS: SetDisplayFactor: factor=0.0500
Mar 31 16:56:28 Jacobs-iPad backboardd[31] <Notice>: ALS: SetDisplayFactor: factor=0.0000
Mar 31 16:56:28 Jacobs-iPad kernel[0] <Debug>: AppleMultitouchN1SPI: updating power statistics
Mar 31 16:56:28 Jacobs-iPad SpringBoard[17] <Warning>: [MPUSystemMediaControls] Disabling lock screen media controls updates for screen turning off.
Mar 31 16:56:28 Jacobs-iPad backboardd[31] <Notice>: MultitouchHID: detection mode: 3->255
Mar 31 16:56:28 Jacobs-iPad SpringBoard[17] <Warning>: [MPUNowPlayingController] Not registered for now playing notifications. Ignoring call to -unregisterForNotifications.
Mar 31 16:56:28 Jacobs-iPad backboardd[31] <Notice>: Posting 'com.apple.iokit.hid.displayStatus' notifyState=0
Mar 31 16:56:28 Jacobs-iPad profiled[2195] <Notice>: (Note ) profiled: Service starting...
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: ALS: AppleARMBacklight::setBacklightEnableGated 0 (set level to 0x25f)
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: AppleMobileADBE0::do_power_state_change(): fSoft: 0 fHard: 1 swapBusy: 0 pendingSwap: 0x0 fController: 1 -> 0 fDisplay: 1 -> 0
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: ALS: AppleARMBacklight::handleMessageGated - framebufferState -> 0
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: 000488.766827 IOAV[2750,280] AppleDPDisplay::setLCDStateGated:(0xffffff808ce85600) Disabling
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: 000000.000130 IOAV[2751,280] AppleDPDisplay::deviceTerminated:(0xffffff808ce85600) device=0xffffff808feb0800 _dpDevice=0xffffff808feb0800
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: 000000.053208 IOAV[2752,280] AppleDPDisplay::terminateLink:(0xffffff808ce85600) Waited 53 ms for display to disable
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: 000000.000087 IOAV[2753,280] AppleDPDisplay::setPower:(0xffffff808ce85600) enable=0 ret=0x00000000
Mar 31 16:56:29 Jacobs-iPad kernel[0] <Debug>: 000000.011656 IOAV[2754,280] AppleDPDisplay::setLCDStateGated:(0xffffff808ce85600) enable=0 ret=0x00000000
Mar 31 16:56:29 Jacobs-iPad backupd[2196] <Warning>: INFO: Scheduling next backup at 4/1/14, 9:58:22 AM
Mar 31 16:56:39 Jacobs-iPad profiled[2195] <Notice>: (Note ) profiled: Service stopping.
Mar 31 16:57:14 Jacobs-iPad backboardd[31] <Notice>: ALS: SetDisplayFactor: factor=1.0000
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: AppleMobileADBE0::do_power_state_change(): fSoft: 1 fHard: 1 swapBusy: 0 pendingSwap: 0x0 fController: 0 -> 1 fDisplay: 0 -> 1
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000045.839491 IOAV[2755,280] AppleDPDisplay::setLCDStateGated:(0xffffff808ce85600) Enabling
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.001375 IOAV[2756,280] AppleDPDisplay::setPower:(0xffffff808ce85600) setting bl deadline for 193 ms from now
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.000006 IOAV[2757,280] AppleDPDisplay::setPower:(0xffffff808ce85600) enable=1 ret=0x00000000
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.000004 IOAV[2758,280] AppleDPDisplay::setLCDStateGated:(0xffffff808ce85600) enable=1 ret=0x00000000
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.022102 IOAV[2759,734609] AppleDPDisplay::devicePublished:(0xffffff808ce85600) newDevice=0xffffff808d390400
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.006347 IOAV[2760,734609] AppleDPDisplay::trainLink:(0xffffff808ce85600) Waited 6 ms for link training
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.000089 IOAV[2761,734609] AppleDPDisplay::initiateLink:(0xffffff808ce85600) Waited 29 ms for display to enable
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.000005 IOAV[2762,734609] AppleDPDisplay::initiateLink:(0xffffff808ce85600) device=0xffffff808d390400 ret=0x00000000
Mar 31 16:57:14 Jacobs-iPad kernel[0] <Debug>: 000000.000005 IOAV[2763,734609] AppleDPDisplay::devicePublished:(0xffffff808ce85600) ret=0x00000000 opened=1
Mar 31 16:57:14 Jacobs-iPad backboardd[31] <Notice>: Posting 'com.apple.iokit.hid.displayStatus' notifyState=1
Mar 31 16:57:14 Jacobs-iPad SpringBoard[17] <Warning>: [MPUSystemMediaControls] Enabling lock screen media controls updates for screen turning on.
Mar 31 16:57:14 Jacobs-iPad SpringBoard[17] <Warning>: [MPUNowPlayingController] Already registered for now playing notifications.
Mar 31 16:57:14 Jacobs-iPad backboardd[31] <Notice>: MultitouchHID: detection mode: 255->6 (deferring until bootloaded)
Mar 31 16:57:15 Jacobs-iPad backboardd[31] <Notice>: MultitouchHID: device bootloaded
Mar 31 16:57:15 Jacobs-iPad backboardd[31] <Notice>: MultitouchHID: detection mode: 6->6
Mar 31 16:57:15 Jacobs-iPad kernel[0] <Debug>: ALS: AppleARMBacklight::handleMessageGated - framebufferState -> 1
Mar 31 16:57:15 Jacobs-iPad kernel[0] <Debug>: ALS: AppleARMBacklight::setBacklightEnableGated 1 (set level to 0x2d3)
Mar 31 16:57:17 Jacobs-iPad backboardd[31] <Notice>: MultitouchHID: detection mode: 6->3
Mar 31 16:57:17 Jacobs-iPad wirelessproxd[2108] <Notice>: (Note ) Created WPTransfer Object for touchsetupd with daemon <DaemonBluetooth: 0x126d0a7d0>
Mar 31 16:57:20 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: installing placeholder for com.testbench.TextView-LayoutManager-Bug
Mar 31 16:57:20 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 handle_install_for_ls: Install of "/var/mobile/Library/Caches/com.apple.itunesstored/AppPlaceholders/-8337678227340918792.app" requested by itunesstored
Mar 31 16:57:20 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 MobileInstallationInstall_Server: Installing app com.testbench.TextView-LayoutManager-Bug
Mar 31 16:57:20 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 install_application: Installing placeholder
Mar 31 16:57:20 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 MobileInstallationInstall_Server: Staging: 0.01s; Waiting: 0.00s; Installation: 0.18s; LS Sync: 0.00s; Overall: 0.25s
Mar 31 16:57:20 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: Creating installProgressForApplication:LSApplicationProxy: com.testbench.TextView-LayoutManager-Bug (Placeholder) withPhase:0
Mar 31 16:57:20 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: Updating installPhase for parent <NSProgress: phase=Loading; state=Waiting; fractionCompleted=0.000000> to 0
Mar 31 16:57:20 Jacobs-iPad itunesstored[2126] <Error>: TASK-ASSERT: cfurlcache - ProcessCacheTask - FAILED to get task-assertion, going commando with 1 items to process.
Mar 31 16:57:20 Jacobs-iPad SpringBoard[17] <Warning>: Killing com.testbench.TextView-LayoutManager-Bug for app installation
Mar 31 16:57:20 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: installing app for existing placeholder LSApplicationProxy: com.testbench.TextView-LayoutManager-Bug (Placeholder)
Mar 31 16:57:20 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: Creating installProgressForApplication:LSApplicationProxy: com.testbench.TextView-LayoutManager-Bug (Placeholder) withPhase:1
Mar 31 16:57:20 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: Updating installPhase for parent <NSProgress: phase=Loading; state=Active; fractionCompleted=0.000000> to 1
Mar 31 16:57:20 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 handle_install_for_ls: Install of "/var/mobile/Media/Downloads/-8337678227340918792/2901536608914947266" requested by itunesstored
Mar 31 16:57:21 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 MobileInstallationInstall_Server: Installing app com.jacobklapper.My-App
Mar 31 16:57:21 Jacobs-iPad installd[2105] <Error>: 0x10050c000 verify_signer_identity: MISValidateSignatureAndCopyInfo failed for /var/tmp/install_staging.ZroEKl/foo_extracted/Payload/My App.app/My App: 0xe8008017
Mar 31 16:57:21 Jacobs-iPad installd[2105] <Error>: 0x10050c000 do_preflight_verification: Could not verify executable at /var/tmp/install_staging.ZroEKl/foo_extracted/Payload/My App.app
Mar 31 16:57:21 Jacobs-iPad installd[2105] <Error>: 0x10050c000 install_application: Could not preflight application install
Mar 31 16:57:21 Jacobs-iPad itunesstored[2126] <Error>: 0x1021b4000 MobileInstallationInstallForLaunchServices: failed with -1
Mar 31 16:57:21 Jacobs-iPad itunesstored[2126] <Warning>: ERROR: MobileInstallationInstallForLaunchServices returned nil
Mar 31 16:57:21 Jacobs-iPad installd[2105] <Error>: 0x10050c000 handle_install_for_ls: API failed
Mar 31 16:57:21 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: installation failed for app com.testbench.TextView-LayoutManager-Bug
Mar 31 16:57:21 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: installPhaseFinishedForProgress: com.testbench.TextView-LayoutManager-Bug.Installing - <NSProgress: phase=Installing; state=Failed; fractionCompleted=0.000000> called, removing progress from cache
Mar 31 16:57:21 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: installation failed for app com.testbench.TextView-LayoutManager-Bug
Mar 31 16:57:21 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: installPhaseFinishedForProgress: com.testbench.TextView-LayoutManager-Bug.Loading - <NSProgress: phase=Loading; state=Failed; fractionCompleted=1.000000> called, removing progress from cache
Mar 31 16:57:24 Jacobs-iPad locationd[58] <Notice>: Location icon should now be in state 'Active'
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Error>: TASK-ASSERT: cfurlcache - ProcessCacheTask - FAILED to get task-assertion, going commando with 1 items to process.
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: Creating installProgressForApplication:LSApplicationProxy: com.testbench.TextView-LayoutManager-Bug (Placeholder) withPhase:0
Mar 31 16:57:28 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: Updating installState for parent <NSProgress: phase=Installing; state=Failed; fractionCompleted=0.000000> to LSInstallStateWaiting
Mar 31 16:57:28 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: Updating installPhase for parent <NSProgress: phase=Installing; state=Waiting; fractionCompleted=0.000000> to 0
Mar 31 16:57:28 Jacobs-iPad SpringBoard[17] <Warning>: Killing com.testbench.TextView-LayoutManager-Bug for app installation
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: installing app for existing placeholder LSApplicationProxy: com.testbench.TextView-LayoutManager-Bug (Placeholder)
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: Creating installProgressForApplication:LSApplicationProxy: com.testbench.TextView-LayoutManager-Bug (Placeholder) withPhase:1
Mar 31 16:57:28 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: Updating installPhase for parent <NSProgress: phase=Loading; state=Active; fractionCompleted=0.000000> to 1
Mar 31 16:57:28 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 handle_install_for_ls: Install of "/var/mobile/Media/Downloads/-8337678227340918792/2901536608914947266" requested by itunesstored
Mar 31 16:57:28 Jacobs-iPad installd[2105] <Notice>: 0x10050c000 MobileInstallationInstall_Server: Installing app com.jacobklapper.My-App
Mar 31 16:57:28 Jacobs-iPad installd[2105] <Error>: 0x10050c000 verify_signer_identity: MISValidateSignatureAndCopyInfo failed for /var/tmp/install_staging.n4trGc/foo_extracted/Payload/My App.app/My App: 0xe8008017
Mar 31 16:57:28 Jacobs-iPad installd[2105] <Error>: 0x10050c000 do_preflight_verification: Could not verify executable at /var/tmp/install_staging.n4trGc/foo_extracted/Payload/My App.app
Mar 31 16:57:28 Jacobs-iPad installd[2105] <Error>: 0x10050c000 install_application: Could not preflight application install
Mar 31 16:57:28 Jacobs-iPad installd[2105] <Error>: 0x10050c000 handle_install_for_ls: API failed
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Error>: 0x1021b4000 MobileInstallationInstallForLaunchServices: failed with -1
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Warning>: ERROR: MobileInstallationInstallForLaunchServices returned nil
Mar 31 16:57:28 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: installation failed for app com.testbench.TextView-LayoutManager-Bug
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: installPhaseFinishedForProgress: com.testbench.TextView-LayoutManager-Bug.Installing - <NSProgress: phase=Installing; state=Failed; fractionCompleted=0.000000> called, removing progress from cache
Mar 31 16:57:28 Jacobs-iPad lsd[1563] <Warning>: LaunchServices: installation failed for app com.testbench.TextView-LayoutManager-Bug
Mar 31 16:57:28 Jacobs-iPad itunesstored[2126] <Warning>: LaunchServices: installPhaseFinishedForProgress: com.testbench.TextView-LayoutManager-Bug.Loading - <NSProgress: phase=Loading; state=Failed; fractionCompleted=1.000000> called, removing progress from cache
Mar 31 16:57:35 Jacobs-iPad locationd[58] <Notice>: Location icon should now be in state 'Region Monitoring'
```

**Comment by SquaredTiki** [comment]
Thanks! The key lines here are:

```
installd[2105] : 0x10050c000 do_preflight_verification: Could not verify executable at /var/tmp/install_staging.n4trGc/foo_extracted/Payload/My App.app
installd[2105] : 0x10050c000 install_application: Could not preflight application install
```

This normally occurs when something isn't quite right about the provisioning profile that is being used (you'll find quite a bit about those errors on Stack Overflow). The usual things to do in this case is to first make sure that the device ID is one which is included in the provisioning profile being used. If that's already the case, re-download and re-install the profile both to your Mac and your iPad. Another is to ensure that Entitlements.plist in the project is correct.

**Comment by mbcoder17** [comment]
Thanks, I'll give it a shot!

**Comment by mbcoder17** [comment]
Hmmmm I tried what you said but I am still getting the same error....is there anything else I can try?

**Comment by SquaredTiki** [comment]
Does it install correctly when running from Xcode onto the device?

**Comment by mbcoder17** [comment]
Yes

**Comment by SquaredTiki** [comment]
I assume your using the same provisioning profile in Xcode as you are in The Constructor. Are all your provisioning profiles and certificates (in Keychain Access) up to date and have not expired?

**Comment by mbcoder17** [comment]
Yup, same provisioning profile...I just deleted any revoked or expired certificates so let me try again...

**Comment by mbcoder17** [comment]
Nope, not working :(

**Comment by SquaredTiki** [comment]
Hmmmm. Do you have multiple provisioning profiles with the same name? Also when you say you are using the team profile, is that a team profile with the wildcard AppID?

**Comment by mbcoder17** [comment]
Nope, none with the same name. Yes, with the wildcard AppID.

**Comment by SquaredTiki** [comment]
And the provisioning profiles on the iPad under Settings > General > Profiles are all correct and the newest ones? Also have you restarted The Constructor after carrying out these suggestions?

**Comment by mbcoder17** [comment]
Yes. I just restarted the constructor and the project is compiling now. Fingers crossed!

**Comment by mbcoder17** [comment]
Nope :(

**Comment by SquaredTiki** [comment]
This is a rather mysterious problem. I will look into this more and see what other details I can find about the errors in question and get back to you as soon as I can. Apologies for these issues.

**Comment by mbcoder17** [comment]
I hope it's not something stupid I'm doing/missing on my end lol. I'm going to try with some different projects and see if they install. If there's anything you need to try and find the problem just let me know.

**Comment by SquaredTiki** [comment]
If you could follow these instructions it might be able to shed some more light on the issue.
1. Go into 'Terminal' and execute the following command
   
   ```
   defaults write com.squaredtiki.The-Constructor KeepProjectWorkingDirectory -boolean YES
   ```
2. Attempt to run your program from your iPad and monitor the console on the Mac, looking for a log which tells you where the IPA file has been exported to, it should be somewhere in a sub-directory in Application Support. The log will look something like this
   
   ```
   The Constructor[12435]: <Builder> Archive and export complete, exported to: /Users/mbcode17/Library/Application Support/The Constructor/5584E3EE-8663-4E62-AA95-B801A92733EE/Export.ipa
   ```
3. Copy the IPA file which should be entitled something along the lines of 'Export.ipa' to your Desktop
4. Open Xcode and then open the Organizer window
5. Switch to the 'Applications' section in the sidebar under the name of your device (device must be plugged in) 
6. Click the 'Add' button and then select the IPA you dragged to your desktop earlier
7. Xcode will then attempt to install the IPA. Wait and see if the IPA installs successfully or if any errors appear.

**Comment by mbcoder17** [comment]
Were there supposed to be .ipa files in the folders before I typed that into terminal?

**Comment by SquaredTiki** [comment]
Nope, only afterwards.

**Comment by mbcoder17** [comment]
Alright, I will try it when I get home.

**Comment by mbcoder17** [comment]
Ok so when I try to add the ipa file xcode gives me this error: "A signed resource has been added, modified, or deleted."

**Comment by SquaredTiki** [comment]
If you've run the same project/app from Xcode before can you try deleting the derived data for it then attempting to install the application to the device again. Making sure the app isn't already installed on the device.

**Comment by mbcoder17** [comment]
I haven't run this project from xcode yet....

**Comment by SquaredTiki** [comment]
Is this a different project to earlier then which you said worked in Xcode?

**Comment by mbcoder17** [comment]
Yes, should I try it with the same one?

**Comment by SquaredTiki** [comment]
If you could, just to see if this result is consistent.

**Comment by mbcoder17** [comment]
Same error from xcode....

**Comment by SquaredTiki** [comment]
Okay, this is interesting. Have a go at this please:
1. Make a copy of the IPA you copied to your desktop earlier
2. Rename it so that it has a `.zip` extension rather than `.ipa`
3. Double-click on the file ('Export.zip') to extract it
4. Open up the Payload folder
5. Open up `Terminal.app` and begin typing the command as follows `codesign -vv`. Then after this drag the `.app` file in the Payload folder into the Terminal window so that the path is inserted resulting in a command a bit like this:
   
   ```
   codesign -vv /Users/mbcoder17/Desktop/Payload/MyApp.app
   ```
6. Press enter to run the command and note the output 

It's late where I am (UK) currently so I will take a look at the output in the morning. Thanks very much for your help in debugging this issue.

**Comment by mbcoder17** [comment]
EDIT: typed it into Terminal wrong. Here is the output:

/Users/Jacob/Desktop/Payload/FinalApp.app: a sealed resource is missing or invalid
file missing: /Users/Jacob/Desktop/Payload/FinalApp.app/archived-expanded-entitlements.xcent

**Comment by SquaredTiki** [comment]
Make sure you leave a space after `-vv` and the path so that the command looks a bit like this:

```
codesign -vv /Users/mbcoder17/Desktop/Payload/MyApp.app
```

**Comment by mbcoder17** [comment]
EDIT: typed it into Terminal wrong. Here is the output:

/Users/Jacob/Desktop/Payload/FinalApp.app: a sealed resource is missing or invalid
file missing: /Users/Jacob/Desktop/Payload/FinalApp.app/archived-expanded-entitlements.xcent

**Comment by SquaredTiki** [comment]
Thanks, that sounds like it could lead to some answers. I will look into that and get back to you once I know more.

**Comment by mbcoder17** [comment]
Great, thanks!

**Comment by SquaredTiki** [comment]
It seems that `archived-expanded-entitlements.xcent` being missing is quite rare. A few questions: what version of Xcode are you running on your Mac? Did the app you were trying to run have a entitlements file? Does this issue occur for all apps even new ones from templates? Have you ensured that the code signing identity you've entered into The Constructor matches that in Keychain Access?

**Comment by mbcoder17** [comment]
Xcode version 5.1
I do not believe I saw an entitlements file in the app. All of the apps I have been trying to run recently have been newly created from templates. Yes, the code signing identities match, but I can check again to make sure.

**Comment by SquaredTiki** [comment]
That's curious, for me at least, building the template projects does not create an `archived-expanded-entitlements.xcent` in the .app file and `codesign -vv` does not expect a `archived-expanded-entitlements.xcent` to exist either. If you could check the code signing identity names definitely match and are in the format `iPhone Developer: Joshua Garnham (HDY7KSGU2Q)` that would be great.

**Comment by SquaredTiki** [comment]
Also could you please upload an IPA which shows the error you are getting with `codesign -vv` so I can take a look myself too please.

**Comment by markmanes** [comment]
Not sure this is relevant but I have been following this conversation -- it might be worth also capturing how his capabilities are set in the project settings. I have seen strange issues with entitlements.plist and code signing when I turned on any of these settings.

Sometimes these switches work in mysterious ways.

-mark=

**Comment by SquaredTiki** [comment]
That was my initial thought but as he is running only projects 'newly created from templates' then no changes should have been made to the capabilities. Something odd is definitely going on though as Xcode should't be creating an `archived-expanded-entitlements.xcent` file I don't believe unless some entitlements are set (by way of the switches in Xcode).

**Comment by markmanes** [comment]
Would it help if he just gave you his project file minus his source?

-mark=

**Comment by SquaredTiki** [comment]
Good point. Seeing as though it's a template project with no modifications could you (mbcoder17) also send me the project (found in the same folder as the IPA) along with the IPA please?

**Comment by mbcoder17** [comment]
Sure, I'll send it over as soon as I get home. Do you want me to create a new project and send it over too, or just the one we've been working with?

**Comment by SquaredTiki** [comment]
Thanks. The one you've been working with so far that you've been providing the logs for so far would be best, thanks again.

**Comment by mbcoder17** [comment]
Here is a zip file of the entire folder: https://www.dropbox.com/s/bjasmzz6khe7zn1/FF59D48A-E555-4B05-899B-6296D344F241.zip

**Comment by SquaredTiki** [comment]
Thanks. First thing I've noticed, do you have two certificates on your system; one set to expire in 78 days and the other in 227 days but both with the same name?

**Comment by mbcoder17** [comment]
Huh, look at that...trying to run it again now....

**Comment by mbcoder17** [comment]
It worked....thanks! I feel kinda stupid now haha. I don't know why I didn't see the duplicate earlier...

**Comment by SquaredTiki** [comment]
Brilliant! No worries at all, I'm just glad we got this fixed 😃

**Comment by mbcoder17** [comment]
Thanks for all your help, appreciate it!
